app/static/js/slcNewProposal.js

```diff
@@ -118,25 +118,39 @@ $(document).ready(function(e) {
       });
 
   // Add course instructor event handlers
-  // -----------------------------------------
       $("#instructorTable").on("click", "#remove", function() {
-        let closestRow =  $(this).closest("tr")
-        $("#instructorTableNames input[value="+closestRow.data('username')+"]").remove()
+        let closestRow = $(this).closest("tr");
+        let username = closestRow.data('username');
+        if (username) {
+            $("#instructorTableNames input[value="+closestRow.data('username')+"]").remove()
+        }
         closestRow.remove();
-      });
-      $("#courseInstructor").on('input', function() {
-          searchUser("courseInstructor", createNewRow, true, null, "instructor");
-      });
+    });
+  // -----------------------------------------
+  $("#instructorTable").on("click", ".removeButton", function() {
+    let closestRow = $(this).closest("tr");
+    let username = closestRow.data('username');
 
-      // for each row in instructorTable that has an instructor, pass that instructors phone data to setupPhoneNumber
-      $('#instructorTable tr').each(function(){
-        var username = getRowUsername(this)
-        var edit = "#editButton-" + username
-        var input = "#inputPhoneNumber-" + username
-        if (username){
-          setupPhoneNumber(edit, input)
-        }
-      })
+    // Check if the username is not empty or undefined
+    if (username) {
+        $("#instructorTableNames input[value='" + username + "']").remove();
+        closestRow.remove();
+    }
+});
+
+    $("#courseInstructor").on('input', function() {
+        searchUser("courseInstructor", createNewRow, true, null, "instructor");
+    });
+
+    // for each row in instructorTable that has an instructor, pass that instructors phone data to setupPhoneNumber
+    $('#instructorTable tr').each(function(){
+      var username = getRowUsername(this)
+      var edit = "#editButton-" + username
+      var input = "#inputPhoneNumber-" + username
```

Comment:
Make a more specific variable name; I think 'input' is not  a so specific variable. Same thing for 'edit' variable, both needs to be more specific to what they are inputting or editing.